### ISSUE TITLE
How to exchange gmail one-time authorization code for access and refresh tokens?

### ISSUE THREAD

**Issue by kramer65** [lepture/flask-oauthlib#237]
**TL;DR**: in the example on [this page](http://oauthlib.readthedocs.org/en/latest/oauth2/grants/authcode.html) I see a line saying `from your_validator import your_validator`. What is that and how can I get it? Does anybody have an example how to exchange a one-time authorization code for an access and a refresh token with flask_oauthlib?

**Full question**
I've been using oauthlib to authenticate people with Gmail to obtain an access and a refresh token in my Flask website successfully like this:
```
from flask_oauthlib.client import OAuth
oauthManager = OAuth(app)

google = oauthManager.remote_app(
    'google',
    consumer_key='xxx.apps.googleusercontent.com',
    consumer_secret='xxx
    request_token_params={
        'scope': ['https://www.googleapis.com/auth/userinfo.email', 'https://www.googleapis.com/auth/gmail.readonly'],
        'access_type': 'offline'
    },
    base_url='https://www.googleapis.com/oauth2/v1/',
    request_token_url=None,
    access_token_method='POST',
    access_token_url='https://accounts.google.com/o/oauth2/token',
    authorize_url='https://accounts.google.com/o/oauth2/auth'
)

@webviewsBlueprint.route('/auth/gmail')
def gmailInit():
    return google.authorize(callback='http://mydomain.com/auth/gmail/callback')
```

This works perfectly fine. But we now also have an app with an API. And from the app I now get a "one-time authorization code" back ([described here](https://developers.google.com/gmail/api/auth/web-server)) which I supposedly need to exchange for an access and refresh token.

I found a description of how to do this with oauthlib [here](http://oauthlib.readthedocs.org/en/latest/oauth2/grants/authcode.html), but I don't really understand it. It starts off with a line saying `from your_validator import your_validator` and I'm immediately stuck. I found something here about implementing a custom validator, but I think that flask_oauthlib should've already implemented this.

Does anybody have any idea what I need to use for this "Validator"? All tips are welcome!

**Comment by mu4farooqi** [comment]
@kramer65 May be this example can help you https://github.com/brunsgaard/yoloAPI.

**Comment by kramer65** [comment]
@mu4farooqi 
Thank you for your suggestion. I've looked through the code of yoloAPI, but I don't think that actually calls an external API with a `one-time authorization code`. Or am I just missing the point of that repo..?

If you have any more tips I'dd be more than happy to hear them.. :-)

**Comment by lepture** [comment]
@mu4farooqi Why do you want to handle one time password yourself? The app will redirect to Google's page for authentication, user will input his one time password at that very page. It has nothing to do with OAuth.

**Comment by mu4farooqi** [comment]
@kramer65  I think @lepture 's answer is for you.

**Comment by lepture** [comment]
@mu4farooqi sorry. It is for @kramer65

**Comment by lepture** [comment]
No response from @kramer65 

Close it.
